File: 1539-diagonal-traverse-ii/diagonal-traverse-ii.java
class Solution {
    public int[] findDiagonalOrder(List<List<Integer>> nums) {
        int size = 0;
        for(var list : nums) {
            size += list.size();
        }
        int[] ans = new int[size];
        int ptr = 0;
        var pq = new PriorityQueue<int[]>((a,b) -> Integer.compare(a[0]+a[1], b[0]+b[1]) == 0 ? Integer.compare(b[0], a[0]) : Integer.compare(a[0]+a[1], b[0]+b[1]));
        for(int i=0; i<nums.size(); i++) {
            for(int j=0; j<nums.get(i).size(); j++) {
                pq.add(new int[]{i, j});
            }
        }
        while(!pq.isEmpty()) {
            ans[ptr++] = nums.get(pq.peek()[0]).get(pq.poll()[1]);
        }
        return ans;
    }
}

File: 1539-diagonal-traverse-ii/diagonal-traverse-ii.py
class Solution:
    def findDiagonalOrder(self, nums: List[List[int]]) -> List[int]:
        q = deque()
        vis = set()
        ans = []

        q.append([0,0])
        vis.add((0,0))
        while q:
            node = q.popleft()
            ans.append(nums[node[0]][node[1]])
            if node[0] + 1 < len(nums) and node[1] < len(nums[node[0]+1]) and (node[0]+1, node[1]) not in vis:
                q.append([node[0]+1, node[1]])
                vis.add((node[0]+1, node[1]))
            if node[1] + 1 < len(nums[node[0]]) and (node[0], node[1]+1) not in vis:
                q.append([node[0], node[1]+1])
                vis.add((node[0], node[1]+1))
        return ans